test/MusicStore.Test/StoreControllerTest.cs

```diff
@@ -0,0 +1,175 @@
+using Microsoft.Framework.DependencyInjection;
+using MusicStore.Models;
+using Xunit;
+
+namespace MusicStore.Controllers
+{
+    public class StoreControllerTest
+    {
+        private readonly IServiceProvider _serviceProvider;
+
+        public StoreControllerTest()
+        {
+            var services = new ServiceCollection();
+
+            services.AddEntityFramework()
+                      .AddInMemoryStore()
+                      .AddDbContext<MusicStoreContext>();
+
+            _serviceProvider = services.BuildServiceProvider();
+        }
+
+        [Fact]
+        public async Task Index_CreatesViewWithGenres()
+        {
+            // Arrange
+            var dbContext = _serviceProvider.GetRequiredService<MusicStoreContext>();
+            CreateTestGenres(numberOfGenres: 10, numberOfAlbums: 1, dbContext: dbContext);
+
+            var controller = new StoreController()
+            {
+                DbContext = dbContext,
+            };
+
+            // Act
+            var result = await controller.Index();
+
+            // Assert
+            var viewResult = Assert.IsType<ViewResult>(result);
+            Assert.Null(viewResult.ViewName);
+
+            Assert.NotNull(viewResult.ViewData);
+            var viewModel = Assert.IsType<List<Genre>>(viewResult.ViewData.Model);
+            Assert.Equal(10, viewModel.Count);
+        }
+
+        [Fact]
+        public async Task Browse_ReturnsHttpNotFoundWhenNoGenreData()
+        {
+            // Arrange
+            var controller = new StoreController()
+            {
+                DbContext = _serviceProvider.GetRequiredService<MusicStoreContext>(),
+            };
+
+            // Act
+            var result = await controller.Browse(string.Empty);
+
+            // Assert
+            Assert.IsType<HttpNotFoundResult>(result);
+        }
+
+        [Fact]
+        public async Task Browse_ReturnsViewWithGenre()
+        {
+            // Arrange
+            var genreName = "Genre 1";
+
+            var dbContext = _serviceProvider.GetRequiredService<MusicStoreContext>();
+            CreateTestGenres(numberOfGenres: 3, numberOfAlbums: 3, dbContext: dbContext);
+
+            var controller = new StoreController()
+            {
+                DbContext = dbContext,
+            };
+
+            // Act
+            var result = await controller.Browse(genreName);
+
+            // Assert
+            var viewResult = Assert.IsType<ViewResult>(result);
+            Assert.Null(viewResult.ViewName);
+
+            Assert.NotNull(viewResult.ViewData);
+            var viewModel = Assert.IsType<Genre>(viewResult.ViewData.Model);
+            Assert.Equal(genreName, viewModel.Name);
+            Assert.NotNull(viewModel.Albums);
+            Assert.Equal(3, viewModel.Albums.Count);
+        }
+
+        [Fact]
+        public async Task Details_ReturnsHttpNotFoundWhenNoAlbumData()
+        {
+            // Arrange
+            var albumId = 0;
```

Comment:
Do we know for sure that `0` will not be generated as id? Instead is it safe to use some negative values?